### PR TITLE
chore: replace protons with protobufjs

### PR DESCRIPTION
**Pull request by achingbrain**
Replaces protons with the official protobufjs module.

### CODE REVIEW

File: test/types/message.spec.js [hunk 1]
```diff
@@ -25,19 +25,14 @@ describe('BitswapMessage', () => {
...
 
-      expect(Message.decode(serialized).wantlist.entries[0]).to.be.eql({
-        block: cid.bytes,
-        priority: 1,
-        cancel: false,
-        sendDontHave: false,
-        wantType: Message.Wantlist.WantType.Block
-      })
+      const deserialized = await BitswapMessage.deserialize(serialized)
+      expect(deserialized.wantlist.get(cid.toString())).to.have.nested.property('entry.wantType', Message.Wantlist.WantType.Block)
```
Comment (achingbrain) [inline test/types/message.spec.js:35]: Protons did some non-standard things here - sendDontHave and wantType should come back as undefined but instead came back as `false` and `0` respectively in line with the comment in the .proto definition.

Protobufjs follows the spec and does not do this so we can't test against the output of the protobuf deserializer but neither should we, really as it's not the unit under test.

Comment (vasco-santos) [reply test/types/message.spec.js:35]: protons was a mess, happy that we are getting rid of it

File: tsconfig.json [hunk 2]
```diff
@@ -5,5 +5,8 @@
   },
   "include": [
     "src"
+  ],
+  "exclude": [
+    "src/types/message/message.js"
```
Comment (vasco-santos) [inline tsconfig.json:10]:
```suggestion
    "src/types/message/message.js" // generated file
```

**Review by vasco-santos — APPROVED** [review]
LGTM!
Pending CI failure on the bundlesize

**Comment by achingbrain** [comment]
Three bytes!

image

**Pull request merged**